pkg/kstatus/status/status.go

```diff
@@ -100,22 +100,30 @@ type Condition struct {
 // a list of standard resources that would belong on the given resource.
 func Compute(u *unstructured.Unstructured) (*Result, error) {
 	res, err := checkGenericProperties(u)
-	if err != nil {
-		return nil, err
-	}
-
 	// If res is not nil, it means the generic checks was able to determine
 	// the status of the resource. We don't need to check the type-specific
 	// rules.
-	if res != nil {
+	if res != nil || err != nil {
```

Comment:
Not sure I understand this addition: `|| err != nil`

Reply:
This ended up including some refactoring that was unrelated to the actual change. I've reverted that part of the PR.